File: back/src/main/java/com/projet/valatMoslehGrodet/dto/CommentDTO.java
package com.projet.valatMoslehGrodet.dto;

import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
@Data
@AllArgsConstructor
@NoArgsConstructor
public class CommentDTO {
    private Long id;
    @NotNull
    private String content;
    @NotNull
    private Integer rating;
    @NotNull
    private String title;
    @NotNull
    private AccountDTO sender;
    @NotNull
    private AccountDTO target;
}

File: front/src/lib/api/models/CommentDto.ts
import { AccountDto } from "./AccountDto";

export type CommentDto = {
  /**
   * @type integer | undefined, int64
   */
  id?: number;
  /**
   * @type string
   */
  content: string;
  /**
   * @type integer, int32
   */
  rating: number;
  /**
   * @type string
   */
  title: string;
  sender: AccountDto;
  target: AccountDto;
};
